zeppelin-web/src/app/notebook/paragraph/result/result.controller.js

```diff
@@ -631,7 +631,7 @@ function ResultCtrl ($scope, $rootScope, $route, $window, $routeParams, $locatio
     let newParagraphConfig = angular.copy(paragraph.config)
     newParagraphConfig.results = newParagraphConfig.results || []
     newParagraphConfig.results[resultIndex] = config
-    if ($scope.revisionView === true) {
+    if ($scope.revisionView === true || $scope.viewOnly) {
```

Comment:
I checked the `$scope.viewOnly` value in this line but it had `false` value. And I checked `true` value when note view mode is `report mode`. 

Thus, I'm not sure that I guess if you want to resolve this problem which is for working `readers` role on the every mode, you refer the following line.
https://github.com/apache/zeppelin/blob/a39ce0141006d5762058b6f7d0cf1ba349d869ac/zeppelin-web/src/app/notebook/notebook.controller.js#L440